tests/backup/backup_helper.go

```diff
@@ -2026,3 +2028,28 @@ func MapToKeyValueString(m map[string]string) string {
 	}
 	return strings.Join(pairs, ",")
 }
+
+// LicenseCountCheck verifies the license count for backup
+func LicenseCountCheck(ctx context.Context, licenseInspectRequestObject *api.LicenseInspectRequest, expectedLicenseCount int64) error {
+	licenseCountCheck := func() (interface{}, bool, error) {
+		licenseInspectResponse, err := Inst().Backup.InspectLicense(ctx, licenseInspectRequestObject)
+		if err != nil {
+			return "", false, err
+		}
+		licenseResponseInfoFeatureInfo := licenseInspectResponse.GetLicenseRespInfo().GetFeatureInfo()
+		if licenseResponseInfoFeatureInfo[0].Consumed == expectedLicenseCount {
+			return "", false, nil
+		}
+		return "", true, nil
+	}
+	_, err := task.DoRetryWithTimeout(licenseCountCheck, licenseCountUpdateTimeout, licenseCountUpdateRetryTime)
+	if err != nil {
+		licenseInspectResponse, err1 := Inst().Backup.InspectLicense(ctx, licenseInspectRequestObject)
+		if err1 != nil {
+			return err1
+		}
+		log.InfoD("Actual license count is %v ,expected license count is %v", licenseInspectResponse.GetLicenseRespInfo().GetFeatureInfo()[0].Consumed, expectedLicenseCount)
+		return err
```

Comment:
Since the `err` returned by `task.Do...` will anyway be something like "context timeout" (which is not useful), you can return the `log.InfoD` statement as error instead (along with `err` embedded in it):
```go
return fmt.Errorf("Actual license count is %v ,expected license count is %v. licenseCountCheck failed with error: %v", licenseInspectResponse.GetLicenseRespInfo().GetFeatureInfo()[0].Consumed, expectedLicenseCount, err)
```

Reply:
corrected